### PR TITLE
Parse time schedule in the event summary

### PR DESCRIPTION
**Pull request by MathieuSchopfer**
Implements feature request #485.
See this [fiddle](https://jsfiddle.net/MathieuSchopfer/0co07f8p/) for an equivalent example.

### CODE REVIEW

File: js/app/controllers/editorcontroller.js [hunk 1]
```diff
@@ -79,6 +79,43 @@ app.controller('EditorController', ['$scope', 'TimezoneService', 'AutoCompletion
...
+				start = moment(start, schemes);
+				if(typeof end !== 'undefined') {
+					end = moment(end, schemes);
+				}
+				else {
+					end = start.clone().add(1, 'hour');
+				}
+				
+				// Update event properties
+				$scope.properties.dtstart.value = moment($scope.properties.dtstart.value.set(
```
Comment (MathieuSchopfer) [inline js/app/controllers/editorcontroller.js]: I would have preferred to directly update the objects `$scope.properties.dtstart/dtend.value` using their [set](http://momentjs.com/docs/#/get-set/set/) method rather creating new ones, but this wouldn't work. I don't understand why ...

File: js/app/controllers/editorcontroller.js [hunk 2]
```diff
@@ -79,6 +79,44 @@ app.controller('EditorController', ['$scope', 'TimezoneService', 'AutoCompletion
...
+				start = moment(start, schemes);
+				if(typeof end !== 'undefined') {
+					end = moment(end, schemes);
+				}
+				else {
+					end = start.clone();
+				}
+				
+				// Update event properties
+				$scope.properties.dtstart.value = moment($scope.properties.dtstart.value.set(
```
Comment (MathieuSchopfer) [inline js/app/controllers/editorcontroller.js:117]: I would have preferred to directly update the objects `$scope.properties.dtstart/dtend.value` using their [set](http://momentjs.com/docs/#/get-set/set/) method rather creating new ones, but this wouldn't work. I don't understand why ...

File: js/app/controllers/editorcontroller.js [hunk 3]
```diff
@@ -79,6 +79,44 @@ app.controller('EditorController', ['$scope', 'TimezoneService', 'AutoCompletion
...
+						{'hour': start.get('hour'), 'minute': start.get('minute')}));
+				$scope.properties.dtend.value = moment($scope.properties.dtend.value.set(
+						{'hour': end.get('hour'), 'minute': end.get('minute')}));
+				$scope.properties.allDay = false;
+				$scope.toggledAllDay();
```
Comment (MathieuSchopfer) [inline js/app/controllers/editorcontroller.js:122]: Call to `$scope.toggledAllDay()` is useful to ensure that the default time interval (1 hour) is defined in just one place.

**Comment by georgehrke** [comment]
cc @jancborchardt

**Comment by jancborchardt** [comment]
Woah nice @MathieuSchopfer! :)

cc @mcorteel @yeKcim please test this as you requested this in #485 

(@MathieuSchopfer if you fix an issue ideally always @-mention the people who were involved in the issue so they see the fix. :)

**Comment by jancborchardt** [comment]
From my side looking at the fiddle it seems good :+1: 
I guess this is mostly something we have to try out in normal use if it works fine.

**Comment by georgehrke** [comment]
will test this tomorrow

**Comment by raghunayyar** [comment]
Please rebase

**Comment by MathieuSchopfer** [comment]
Done ;-)

**Comment by raghunayyar** [comment]
@georgehrke can we have another pair of eyes here? :)

One sec, the diff seems weird though with the public/app.js compilation.

**Comment by georgehrke** [comment]
please remove the app.js 
(it was deleted in master)

**Comment by georgehrke** [comment]
Hm, what if someone want's to have a title like "15:00 title".
It'll always get replaced. :/

**Comment by raimund-schluessler** [comment]
Maybe introduce a setting to turn off automatic parsing? :see_no_evil:

**Comment by georgehrke** [comment]
I would prefer a dropdown that shows different options what to do.
There is an issue about that. Mostly related to entering titles like dinner and automatically suggesting setting the time to 8pm or something similar

**Comment by raimund-schluessler** [comment]
I haven't had time to look at this PR, but I guees since `ng-blur` is used it only updates when the input field loses focus.

Hence, one improvement may be to evaluate the summary on keypress to directly show what the user can expect.
Also, one could think about only parsing the summary in the popover editor but not in the sidebar as a help to quickly create an event. Thus it would still be possible to change the title e.g. to `15:00 title` in the sidebar later on (this is similar to how e.g. Fantastical 2 does it on iOS).

**Pull request closed without merging**